.github/workflows/linux.yml

```diff
@@ -32,56 +32,43 @@ env:
 
 jobs:
   build:
-    runs-on: ubuntu-24.04
+    name: build-${{ matrix.dist }}-${{ matrix.arch }}
+    strategy:
+      matrix:
+        dist: ['22.04', '24.04']
+        arch: ['amd64', 'arm64']
+    runs-on: ubuntu-${{ matrix.dist }}${{ matrix.arch == 'arm64' && '-arm' || '' }}
     steps:
       - uses: actions/checkout@v4
       - uses: awalsh128/cache-apt-pkgs-action@v1.4.3
         with:
           packages: ${{ env.BASE_DEPS }}
-          version: apt-20-base
+          version: apt-${{ matrix.dist }}-${{ matrix.arch }}-base
           execute_install_scripts: true
       - run: .github/setup-linux.sh
-      - run: .github/build.sh dist
+      - run: .github/build.sh ${{ matrix.arch == 'amd64' && 'dist' || '' }} valgrind
       - name: Upload test logs
         uses: actions/upload-artifact@v4
         if: failure()
         with:
-          name: ubuntu-test-logs
+          name: ubuntu-${{ matrix.dist }}-${{ matrix.arch }}-test-logs
           path: |
             tests/*.log
             src/tests/unittests/*.log
       - uses: actions/cache@v4
         id: cache-build
+        if: matrix.arch == 'amd64'
         with:
           path: ./*
-          key: ${{ runner.os }}-${{ github.sha }}
+          key: ${{ runner.os }}-${{ matrix.dist }}-${{ matrix.arch }}-${{ github.sha }}
       - name: Upload build artifacts
+        if: matrix.arch == 'amd64'
         uses: actions/upload-artifact@v4
         with:
-          name: opensc-build
+          name: opensc-build-${{ matrix.dist }}-${{ matrix.arch }}
           path:
             opensc*.tar.gz
 
-  valgrind:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: awalsh128/cache-apt-pkgs-action@v1.4.3
-        with:
-          packages: ${{ env.BASE_DEPS }}
-          version: apt-latest-base
-          execute_install_scripts: true
-      - run: .github/setup-linux.sh debug
-      - run: .github/build.sh valgrind
```

Comment:
The separate valgrind job had the `debug` argument to setup, which installed debuginfo to have reasonable reports of errors. It also runs very slowly, which was a reason why we did not enable it for the main jobs.

Reply:
oh I did not notice, ubuntu 22.04 did not have the debug option

Reply:
Resolved